src/dotnet/UI.Blazor/Components/Modal/DataBag.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace ActualChat.UI.Blazor.Components;
+
+public class DataBag
```

Comment:
looks really weird... This is just a hashtable, isn't it?

Reply:
Removed DataBag class, Added extension methods to IDictionary<string, object> and used it to hold page and modal shared data.